index.html

```diff
@@ -5007,13 +4989,281 @@ <h3 id="looping-AudioBufferSourceNode">
+            the node.
+            </li>
+            <li>Let <code>bufferRate</code> be the sample rate employed by
+            <code>buffer</code>.
+            </li>
+            <li>Define a <em>playhead position</em> for <code>buffer</code> as
+            any quantity representing an exact, unquantized time offset in
+            seconds, relative to the time coordinate of the first sample frame
+            in the buffer. Playback rate and <a>AudioContext</a> sample rate
+            are not relevant: this offset is expressed purely in terms of the
+            buffer's audio content and <code>bufferRate</code>.
+            </li>
+            <li>Let <code>offset</code> correspond to the <code>offset</code>
+            parameter to <code>start()</code> if supplied, else 0. This is the
+            <em>playhead position</em> in the buffer at which playback will
+            begin.
+            </li>
+            <li>Let <code>loop</code>, <code>actualLoopStart</code> and <code>
+              actualLoopEnd</code> be the loop-related attributes of the node,
+              as discussed in the preceding section on looping. Note that the
+              loop start and loop end points are <em>playhead positions</em>,
+              not playback times.
+            </li>
+            <li>Let the function <code>playbackSignal(position)</code> be the
+            <em>playback signal function</em> for <code>buffer</code>, which is
+            a function that maps from a <em>playhead position</em> to a set of
+            output signal values, one for each output channel. If
+            <code>position</code> corresponds to the location of an exact
+            sample frame in the buffer, this function returns that frame.
+            Otherwise, its return value is determined by a UA-supplied
+            algorithm that interpolates between sample frames in the
+            neighborhood of <code>position</code>. If <code>position</code> is
+            greater than or equal to <code>actualLoopEnd</code> and there is no
+            subsequent sample frame in <code>buffer</code>, then the next
+            sample frame should be taken as the playback signal at
+            <code>actualLoopStart</code> for interpolation purposes.
+            </li>
+            <li>Let the operation <em>optimize the buffer</em> be any operation
+            that alters both the buffer contents and sample rate in a way that
+            increases the efficiency or quality of rendering, while minimizing
+            changes to the <code>playbackSignal()</code> function. The nature
+            of this operation is up to the UA. Examples of such operations
+            might include upsampling, downsampling, applying a subsample
+            offset, or loop unrolling. This operation may affect the values of
+            <code>sampleRate</code>, <code>playbackRate</code> and
+            <code>offset</code>.
```

Comment:
I think you're right about sampleRate and offset. But if you resample a buffer, then its sample rate changes, doesn't it? 

Maybe this is moot though if we just fuzz out the whole desription of optimization and let the UA do whatever it wants so long as it's effectively invisible to the outcome.